client/app/components/Input/InputSwitch.jsx

```diff
@@ -0,0 +1,88 @@
+// @flow
+import React from 'react';
+import { Utils } from '../../utils';
+import { Input } from './index';
+import css from './InputSwitch.scss';
+import { I18n } from '../../libs/i18n';
+
+export type Props = {
+  id: string,
+  name: string,
+  label: string,
+  value?: any,
+  checked?: boolean,
+  uncheckedValue?: any,
+};
+
+export type State = {
+  checked: boolean,
+  key?: string,
+};
+
+export class InputSwitch extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { checked: !!props.checked };
```

Comment:
This is actually just a question out of curiosity on my part, not a suggestion or anything! What's the advantage of using `!!` here when that particular prop is defined as a boolean via Flow?

Reply:
Good question, it's because this.props.checked can actually be undefined or a boolean. For the generic Input component, this component is a child of, checked and most of the props are optional. this.state.checked in this particular component is a boolean so I needed a way of converting undefined to false.

Reply:
Ahh, I see, thanks for the detailed explanation! I completely overlooked the `?` in the definition, so I didn't consider the possibility of the prop being undefined.